internal/node/payload_cache.go

```diff
@@ -56,26 +52,12 @@ func (m *MachineManager) CacheWorkload(request *controlapi.DeployRequest) (uint6
 		return 0, nil, err
 	}
 
-	filename := path.Join(os.TempDir(), uuid.NewString())
-	err = store.GetFile(key, filename)
+	workload, err := store.GetBytes(key)
 	if err != nil {
 		m.log.Error("Failed to download bytes from source object store", slog.Any("err", err), slog.String("key", key))
 		return 0, nil, err
 	}
 
-	f, err := os.Open(filename)
-	if err != nil {
-		return 0, nil, err
-	}
-
-	workload, err := io.ReadAll(f)
-	if err != nil {
-		m.log.Error("Couldn't read the file we just wrote", slog.Any("err", err))
-		return 0, nil, err
-	}
-
-	os.Remove(filename)
```

Comment:
Since we deleted the file right after doing `io.ReadAll`, we no longer use the file, as far as I can tell.